### ISSUE TITLE
installing modules with git slugs improperly resolves resource urls

### ISSUE THREAD

**Issue by dstokes** [npm/npm#6422]
```sh
$ npm install -g org/private-module
...

$ npm ls -g
npm ERR! Unsupported URL Type: git+git+ssh://git@github.com:org/private-module#hash
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/stefansundin/npm-debug.log
```
note the `git+git+ssh://`protocol.

installing with the full git ssh url seems to fix the problem:
```sh
$ npm install -g git+ssh://git@github.com:org/private-module.git
```

here's the debug log
```log
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'ls', '-g' ]
2 info using npm@2.0.0
3 info using node@v0.10.32
4 verbose node symlink /usr/local/bin/node
6 verbose stack Error: Unsupported URL Type: git+git+ssh://git@github.com:org/private-module#hash
6 verbose stack     at parseUrl (/usr/local/lib/node_modules/npm/node_modules/npm-package-arg/npa.js:173:13)
6 verbose stack     at npa (/usr/local/lib/node_modules/npm/node_modules/npm-package-arg/npa.js:74:12)
6 verbose stack     at makeArchy_ (/usr/local/lib/node_modules/npm/lib/ls.js:281:9)
6 verbose stack     at /usr/local/lib/node_modules/npm/lib/ls.js:296:14
6 verbose stack     at Array.map (native)
6 verbose stack     at makeArchy_ (/usr/local/lib/node_modules/npm/lib/ls.js:295:22)
6 verbose stack     at makeArchy (/usr/local/lib/node_modules/npm/lib/ls.js:228:13)
6 verbose stack     at /usr/local/lib/node_modules/npm/lib/ls.js:65:13
6 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-installed/read-installed.js:138:5
6 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-installed/read-installed.js:251:14
7 verbose cwd /Users/user
8 error Darwin 13.4.0
9 error argv "node" "/usr/local/bin/npm" "ls" "-g"
10 error node v0.10.32
11 error npm  v2.0.0
12 error Unsupported URL Type: git+git+ssh://git@github.com:org/private-module#hash
13 error If you need help, you may report this error at:
13 error     <http://github.com/npm/npm/issues>
14 verbose exit [ 1, true ]
```

**Comment by gaggle** [comment]
Just to say I've upgraded to

    $ npm --version
    2.1.18`

I still get the error though:

    $ npm -g list
    npm ERR! Darwin 14.0.0
    npm ERR! argv "node" "/usr/local/bin/npm" "-g" "list"
    npm ERR! node v0.10.35
    npm ERR! npm  v2.1.18

    npm ERR! Unsupported URL Type: git+git+ssh://git@github.com:org/private- module#hash
    npm ERR! 
    npm ERR! If you need help, you may report this error at:
    npm ERR!     <http://github.com/npm/npm/issues>

    npm ERR! Please include the following file with any support request:
    npm ERR!     /Users/jla/Projects/foo/npm-debug.log

Looking at the error message I noticed the `git+git+ssh` part. That looks weird. So I went to that module's `package.json` and changed the entry to `git+ssh`. And now `npm -g list` works!

I install that module via:

    $ npm install org/private-module -g
    /usr/local/bin/private-module -> /usr/local/lib/node_modules/private-module/index.js
    private-module@0.0.0 /usr/local/lib/node_modules/private-module
    └── private@0.10.0 (once@1.3.1, amqplib@0.2.1, tape@3.0.3, stats@0.4.1)

npm automatically resolves that to the correct github repo (nice!), but the process also adds meta-data to the module's `package.json`. Specifically a `_resolved` attribute is added that has the `git+git+ssh` specifier. I don't know enough about npm to say for sure that this is actually problematic nor if I actually fixed anything or this just masks a symptom. Maybe I'm barking up the wrong tree. But I won't know until I post it somewhere, so I'm leaving all this data here in case someone might find it useful :)

**Comment by smikes** [comment]
Can this issue now be closed?

From the above discussion, I think this is an old bug that is fixed in current `npm` versions (2.4.1 is `latest`).  If it's not fixed, just let us know and it won't be closed.

We are trying to clean up older npm issues, so if we don't hear back from you within a week, we will close this issue. (Don't worry -- you can always come back again and open a new issue!)

Thanks!

**Comment by gaggle** [comment]
I don't think this is fixed at all.

```
$ npm -v
2.4.1

$ npm list -g
/usr/local/lib
├─┬ private-module@0.0.0 (git+ssh://git@github.com:org/private-module#hash)
[dependencies]

$ npm install org/private-module -g 
/usr/local/bin/private-module -> /usr/local/lib/node_modules/private-module/index.js
private-module@1.0.0 /usr/local/lib/node_modules/private-module
[dependencies]

$ npm list -g
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "list" "-g"
npm ERR! node v0.10.35
npm ERR! npm  v2.4.1
npm ERR! Unsupported URL Type: git+git+ssh://git@github.com:org/private-module#hash
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/user/npm-debug.log
```

To make it work again I `nano /usr/local/lib/node_modules/private-module/package.json` and edit the `_resolved` tag from `"git+git+ssh://url"` to `"git+ssh://url"`. Then `npm list -g` works again.

Please let me know if I can provide further information.

**Comment by KenanY** [comment]
This looks like what https://github.com/npm/npm/pull/7296 fixes? ref. https://github.com/npm/npm/issues/7294

**Comment by othiym23** [comment]
Indeed, this is the same issue that #7296 addresses. As soon as we have a decent test for that, we'll be merging it, so this should get addressed in `npm@2` before too long.

**Comment by othiym23** [comment]
Closing as a duplicate of #7296, which will probably land in the next npm release.

**Comment by othiym23** [comment]
Closed by a2e04bd921feab8f9e40a27e180ca9308eb709d7 and 0f87f5ed28960d962f34977953561d22983da4f9. It turns out that the fact that it only affected private GitHub repos was a very important clue, and allowed us to fix both the proximate error (where private repos were having their URLs saved incorrectly) and the underlying problem (URL normalization for caching Git remotes wasn't general enough). So thanks for pointing this out, as well as your patience!

**Comment by dstokes** [comment]
:D :+1:
